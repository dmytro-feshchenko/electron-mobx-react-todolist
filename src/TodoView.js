import React, { Component } from 'react';

class TodoView extends Component {
    render() {
        const todo = this.props.todo;
        return (
            <li onDoubleClick={ this.onRename }>
                <input
                    type='checkbox'
                    checked={ todo.completed }
                    onChange={ this.onToggleCompleted }
                />
                { todo.text }
                { todo.assignee
                    ? <small>{ todo.assignee.name }</small>
                    : null
                }
            </li>
        )
    }

    /**
     * Change a completed status of the task
     * Sets status as completed if not completed yet.
     * Otherwise, sets a status as not completed.
     */
    onToggleCompleted = () => {
        const todo = this.props.todo;
        todo.completed = !todo.completed;
    }

    /**
     * Calls a prompt for changing the task name
     */
    onRename = () => {
        const todo = this.props.todo;
        todo.text = prompt('Task name', todo.text) || "";
    }
}

export default TodoView;