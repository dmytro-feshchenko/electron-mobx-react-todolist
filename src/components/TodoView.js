import React, { Component } from 'react';

/**
 * TodoView component
 * Represents one task item
 */
class TodoView extends Component {
    /**
     * Constructor for TodoView
     * @param props
     */
    constructor(props) {
        super(props);
        
        this._onRename = this._onRename.bind(this);
        this._onToggleCompleted = this._onToggleCompleted.bind(this);

        this.state = {};
    }

    /**
     * Renders the task item
     * @returns {XML}
     */
    render() {
        const todo = this.props.todo;
        return (
            <li onDoubleClick={ this._onRename }>
                <input
                    type='checkbox'
                    checked={ todo.completed }
                    onChange={ this._onToggleCompleted }
                />
                { todo.text }
                { todo.assignee
                    ? <small>{ todo.assignee.name }</small>
                    : null
                }
            </li>
        )
    }

    /**
     * Change a completed status of the task
     * Sets status as completed if not completed yet.
     * Otherwise, sets a status as not completed.
     */
    _onToggleCompleted() {
        const todo = this.props.todo;
        todo.completed = !todo.completed;
    }

    /**
     * Calls a prompt for changing the task name
     */
    _onRename() {
        const todo = this.props.todo;
        todo.text = prompt('Task name', todo.text) || "";
    }
}

export default TodoView;