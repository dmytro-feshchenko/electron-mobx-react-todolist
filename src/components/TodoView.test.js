import React from 'react';
import ReactDOM from 'react-dom';
import TodoView from './TodoView';
import renderer from 'react-test-renderer';

it("renders correctly for not completed tasks", () => {

    const todo = {
        text: "Reminder to buy milk after work",
        completed: false
    };

    const tree = renderer.create(
        <TodoView todo={todo}/>
    );

    expect(tree).toMatchSnapshot();
});

it("renders correctly for completed tasks", () => {
    const todo = {
        text: "Reminder to buy milk after work",
        completed: true
    };

    const tree = renderer.create(
        <TodoView todo={todo}/>
    );

    expect(tree).toMatchSnapshot();
});

it("renders correctly for tasks with assignee", () => {
    const todo = {
        text: "Reminder to buy milk after work",
        completed: true,
        assignee: {
            name: "John Doe"
        }
    };

    const tree = renderer.create(
        <TodoView todo={todo}/>
    );

    expect(tree).toMatchSnapshot();
});