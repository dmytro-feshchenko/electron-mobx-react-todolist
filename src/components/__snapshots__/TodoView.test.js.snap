// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for completed tasks 1`] = `
<li
  onDoubleClick={[Function]}
>
  <input
    checked={true}
    onChange={[Function]}
    type="checkbox"
  />
  Reminder to buy milk after work
</li>
`;

exports[`renders correctly for not completed tasks 1`] = `
<li
  onDoubleClick={[Function]}
>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  Reminder to buy milk after work
</li>
`;

exports[`renders correctly for tasks with assignee 1`] = `
<li
  onDoubleClick={[Function]}
>
  <input
    checked={true}
    onChange={[Function]}
    type="checkbox"
  />
  Reminder to buy milk after work
  <small>
    John Doe
  </small>
</li>
`;
